profitbricks/src/main/java/org/jclouds/profitbricks/features/SnapshotApi.java

```diff
@@ -0,0 +1,86 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.profitbricks.features;
+
+import java.util.List;
+
+import javax.inject.Named;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.jclouds.Fallbacks;
+import org.jclouds.http.filters.BasicAuthentication;
+import org.jclouds.profitbricks.binder.snapshot.CreateSnapshotRequestBinder;
+import org.jclouds.profitbricks.binder.snapshot.RollbackSnapshotRequestBinder;
+import org.jclouds.profitbricks.binder.snapshot.UpdateSnapshotRequestBinder;
+import org.jclouds.profitbricks.domain.Snapshot;
+import org.jclouds.profitbricks.http.filters.ProfitBricksSoapMessageEnvelope;
+import org.jclouds.profitbricks.http.parser.RequestIdOnlyResponseHandler;
+import org.jclouds.profitbricks.http.parser.snapshot.SnapshotResponseHandler;
+import org.jclouds.profitbricks.http.parser.snapshot.SnapshotListResponseHandler;
+import org.jclouds.rest.annotations.Fallback;
+import org.jclouds.rest.annotations.MapBinder;
+import org.jclouds.rest.annotations.Payload;
+import org.jclouds.rest.annotations.PayloadParam;
+import org.jclouds.rest.annotations.RequestFilters;
+import org.jclouds.rest.annotations.XMLResponseParser;
+
+@RequestFilters({BasicAuthentication.class, ProfitBricksSoapMessageEnvelope.class})
+@Consumes(MediaType.TEXT_XML)
+@Produces(MediaType.TEXT_XML)
+public interface SnapshotApi {
+
+   @POST
+   @Named("snapshot:getall")
+   @Payload("<ws:getAllSnapshots/>")
+   @XMLResponseParser(SnapshotListResponseHandler.class)
+   @Fallback(Fallbacks.EmptyListOnNotFoundOr404.class)
+   List<Snapshot> getAllSnapshots();
+
+   @POST
+   @Named("snapshot:get")
+   @Payload("<ws:getSnapshot><snapshotId>{snapshotId}</snapshotId></ws:getSnapshot>")
+   @XMLResponseParser(SnapshotResponseHandler.class)
+   @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+   Snapshot getSnapshot(@PayloadParam("snapshotId") String identifier);
+
+   @POST
+   @Named("snapshot:create")
+   @MapBinder(CreateSnapshotRequestBinder.class)
+   @XMLResponseParser(SnapshotResponseHandler.class)
+   Snapshot createSnapshot(@PayloadParam("snapshot") Snapshot.Request.CreatePayload payload);
+
+   @POST
+   @Named("snapshot:update")
+   @MapBinder(UpdateSnapshotRequestBinder.class)
+   @XMLResponseParser(RequestIdOnlyResponseHandler.class)
+   String updateSnapshot(@PayloadParam("snapshot") Snapshot.Request.UpdatePayload payload);
+
+   @POST
+   @Named("snapshot:delete")
+   @Payload("<ws:deleteSnapshot><snapshotId>{id}</snapshotId></ws:deleteSnapshot>")
+   @Fallback(Fallbacks.FalseOnNotFoundOr404.class)
+   boolean deleteSnapshot(@PayloadParam("id") String id);
+
+   @POST
+   @Named("snapshot:rollback")
+   @MapBinder(RollbackSnapshotRequestBinder.class)
+   boolean rollbackSnapshot(@PayloadParam("snapshot") Snapshot.Request.RollbackPayload payload);
```

Comment:
So, if I understand this properly, this method will never return false. If it suceeds returns "true", and fails with an exception otherwise. It sounds better to me to return the requestId or just `void`. WDYT?

Regarding the delete method: does the API actually return a 404? If the API never returns a 404, then remove the fallback and apply the same criteria applied to the rollback method.